src/Stratis.FederatedPeg.Features.FederationGateway/Controllers/FederationGatewayController.cs

```diff
@@ -162,25 +174,46 @@ public IActionResult ReceiveCurrentBlockTip([FromBody] BlockTipModelRequest bloc
         }
 
         /// <summary>
-        /// Trigger a re-sync of missing matured blocks deposits.
+        /// Trigger a sync of missing matured blocks deposits.
         /// </summary>
-        /// <param name="maturedBlockTip"><see cref="MaturedBlockModel"/>Last known Block tip Hash and Height.</param>
+        /// <param name="blockHashHeight">Last known Block tip Hash and Height.</param>
         /// <returns><see cref="IActionResult"/>OK on success.</returns>
         [Route(FederationGatewayRouteEndPoint.ReSyncMaturedBlockDeposits)]
         [HttpPost]
-        public IActionResult ResyncMaturedBlockDeposits([FromBody] MaturedBlockModel maturedBlockTip)
+        public IActionResult ResyncMaturedBlockDeposits([FromBody] MaturedBlockModel blockHashHeight)
         {
-            Guard.NotNull(maturedBlockTip, nameof(maturedBlockTip));
+            Guard.NotNull(blockHashHeight, nameof(blockHashHeight));
 
             if (!this.ModelState.IsValid)
             {
-                return BuildErrorResponse(this.ModelState);
+                IEnumerable<string> errors = this.ModelState.Values.SelectMany(e => e.Errors.Select(m => m.ErrorMessage));
+                return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, "Formatting error", string.Join(Environment.NewLine, errors));
             }
 
             try
-            {
-                //TODO - Add logic to process mature block deposits for each missing block.                
+            {             
+                ChainedHeader chainedHeader = this.chain.GetBlock(blockHashHeight.BlockHash);
+
+                if (chainedHeader == null)
+                {
+                    return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, $"Block with hash {blockHashHeight.BlockHash} was not found on the blockchain.", string.Empty);
+                }
+
+                int currentHeight = chainedHeader.Height;
+
+                while (currentHeight < this.chain.Tip.Height)
+                {
+                    IMaturedBlockDeposits maturedBlockDeposits =
+                        this.depositExtractor.ExtractMaturedBlockDeposits(chainedHeader);
+
+                    if (maturedBlockDeposits == null) continue;
+
+                    this.maturedBlockSender.SendMaturedBlockDepositsAsync(maturedBlockDeposits).ConfigureAwait(false);
+
+                    currentHeight++;
 
+                    chainedHeader = this.chain.GetBlock(currentHeight);
+                }

```

Comment:
For now the above logic will send each missing `maturedBlockDeposits`.

This is the initial version.  I was thinking to move this into it's own class.  So the call to this controller method will fall through - a bit like WalletSyncManager.  

Then in the class will send the deposits in a separate process.

Also as discussed based on performance we could consider batch the deposit blocks up and sending in bulk.